### ISSUE TITLE
Android: Lifecycle usage (LifecycleEventListener, LifecycleOwner, Lifecycle)

### ISSUE THREAD

**Issue by mrousavy** [facebook/react-native#30728]
## Description

I'm currently creating a native module which requires a [`Lifecycle`](https://developer.android.com/reference/androidx/lifecycle/Lifecycle), and unfortunately there's no way around that (CameraX API constructor only takes `Lifecycle` or `LifecycleOwner` as arguments)

React Native provides a way to listen for Lifecycle events using a `LifecycleEventListener` (`ReactContext::addLifecycleEventListener`), but that's not using the `Lifecycle` interface.

I am really losing my mind trying to figure out how I can obtain an actual `Lifecycle` (or `LifecycleOwner`) instance from React Native, but I can't figure it out. Even creating a custom `Lifecycle` requires using `LifecycleObserver`s which are not the same as the React `LifecycleEventListeners`.

Can someone help me out?

TL;DR: I'm trying to call [`ProcessCameraProvider::bindToLifecycle(lifecycle: Lifecycle)`](https://developer.android.com/reference/androidx/camera/lifecycle/ProcessCameraProvider#bindToLifecycle(androidx.lifecycle.LifecycleOwner,%20androidx.camera.core.CameraSelector,%20androidx.camera.core.UseCase...)) in a React Native native module, but there's no way I can get a `Lifecycle` instance.

**Comment by mrousavy** [comment]
Nevermind, you can just cast the activity to an AppCompatActivity.

```kt
private val lifecycle: Lifecycle by lazy {
    ((context as ReactContext).currentActivity as AppCompatActivity).lifecycle
}
```

It would probably be a good idea to change the type of `.currentActivity` to `AppCompatActivity` if we're not using `Activity` any more.
